source/production/F0.Minesweeper.Components/Minefield.razor.cs

```diff
@@ -0,0 +1,18 @@
+using F0.Minesweeper.Components.Abstractions;
+using Microsoft.AspNetCore.Components;
+
+namespace F0.Minesweeper.Components
+{
+	public partial class Minefield
+	{
+		[Parameter]
+		public MinefieldSize Size { get; set; }
+
+		private bool isValidSize;
+
+		protected override void OnParametersSet()
+		{
+			isValidSize = Size != null && Size.Height > 0 && Size.Width > 0;
```

Comment:
Since `MinefieldSize` is a _record type_, for which the compiler synthesizes a `operator ==`, we should use _pattern matching_ rather than the _equality operator_ to check for _null_. That way, we just check for _null_, rather than detouring via `==` and `Equals`.

```suggestion
			isValidSize = Size is not null && Size.Height > 0 && Size.Width > 0;
```